crates/hir-expand/src/fixup.rs

```diff
@@ -194,7 +193,105 @@ pub(crate) fn fixup_syntax(node: &SyntaxNode) -> SyntaxFixups {
                     }
                 },
                 // FIXME: foo::
-                // FIXME: for, match etc.
+                ast::MatchExpr(it) => {
+                    if it.expr().is_none() {
+                        let match_token = match it.match_token() {
+                            Some(t) => t,
+                            None => continue
+                        };
+                        append.insert(match_token.into(), vec![
+                            SyntheticToken {
+                                kind: SyntaxKind::IDENT,
+                                text: "__ra_fixup".into(),
+                                range: end_range,
+                                id: EMPTY_ID
+                            },
+                        ]);
+                    }
+                    if it.match_arm_list().is_none() {
+                        // No match arms
+                        append.insert(node.clone().into(), vec![
+                            SyntheticToken {
+                                kind: SyntaxKind::L_CURLY,
+                                text: "{".into(),
+                                range: end_range,
+                                id: EMPTY_ID,
+                            },
+                            SyntheticToken {
+                                kind: SyntaxKind::UNDERSCORE,
+                                text: "_".into(),
+                                range: end_range,
+                                id: EMPTY_ID
+                            },
+                            SyntheticToken {
+                                kind: SyntaxKind::EQ,
+                                text: "=".into(),
+                                range: end_range,
+                                id: EMPTY_ID
+                            },
+                            SyntheticToken {
+                                kind: SyntaxKind::R_ANGLE,
+                                text: ">".into(),
+                                range: end_range,
+                                id: EMPTY_ID
+                            },
+                            SyntheticToken {
+                                kind: SyntaxKind::L_CURLY,
+                                text: "{".into(),
+                                range: end_range,
+                                id: EMPTY_ID,
+                            },
+                            SyntheticToken {
+                                kind: SyntaxKind::R_CURLY,
+                                text: "}".into(),
+                                range: end_range,
+                                id: EMPTY_ID,
+                            },
```

Comment:
This isn't necessary, it's enough for us to generate just a `{}` pair for the match arm list here